### PR TITLE
feat/display-name

### PR DESCRIPTION
**Pull request by mansurishmuhametov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/common/integration/tests/pipes/entity-name-pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        pipe = TestBed.inject(NameByPathPipe);
+    });
+
+    it('should return entity name', () => {
+        expect(pipe.transform('a')).toEqual('a');
+        expect(pipe.transform([])).toEqual('');
+        expect(pipe.transform(null)).toEqual('');
+        expect(pipe.transform({ name: 'a' }, 'name')).toEqual('a');
+        expect(pipe.transform({ value: { name: 'a' } }, 'value.name')).toEqual('a');
+        expect(function () {
```
Comment (nikitasadchikov) [inline packages/common/integration/tests/pipes/entity-name-pipe.spec.ts:18]: Why not to use an arrow function? Something like this: `expect(() => ...).toThrow(...)`

File: packages/common/object/src/get-name-by-path.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Any } from '@angular-ru/common/typings';
+import { checkValueIsEmpty } from '@angular-ru/common/utils';
+
+import { getValueByPath } from './get-value-by-path';
+import { isObject } from './is-object';
+
+export function getNameByPath<T>(item: T, displayKey?: keyof T): string {
```
Comment (Schirbak) [inline packages/common/object/src/get-name-by-path.ts:7]: didn't catch:
1. what does the word `name` means in the context of `T` which is string or any abstract object? e.g. `{ opened: true, width: 300, title: 'Some Titile' }` — which of these fields can be called a name?
2. why is there a separate function? i think `getValueByPath` is quite enough for these purposes. i see it insures against falsy values, but there is no need for this. anyway you can use `??` in the pipe.

Comment (splincode) [inline packages/common/object/src/get-name-by-path.ts:7]: @Schirbak I'm a little worried that I don't fully understand the semantics this code

File: packages/common/integration/tests/pipes/entity-name-pipe.spec.ts (hunk 1 above)
Comment (mansurishmuhametov) [reply packages/common/integration/tests/pipes/entity-name-pipe.spec.ts:18]: I will replace it with an arrow function

File: packages/common/object/src/get-name-by-path.ts (hunk 2 above)
Comment (mansurishmuhametov) [reply packages/common/object/src/get-name-by-path.ts]: moved the function logic to the pipe

Comment (mansurishmuhametov) [reply packages/common/object/src/get-name-by-path.ts]: 1. `name` looks a little confusing, so I renamed pipe: `itemName` -> `displayItem`
2. removed `getNameByPath`

**Review by nikitasadchikov — DISMISSED** [review]

**Review by splincode — CHANGES_REQUESTED** [review]

File: packages/common/pipes/src/display-item/display-item.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            return value;
+        }
+    }
+
+    private parseNotObject<T>(item: T): string {
+        if (typeof item === 'string') {
+            return item;
+        }
+        return '';
+    }
```
[the remark this answers is not in the document]
Comment (mansurishmuhametov) [reply packages/common/pipes/src/display-item/display-item.pipe.ts:36]: agree, looks better

File: packages/common/pipes/src/display-item/display-item.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        if (checkValueIsEmpty(displayKey) || typeof displayKey !== 'string') {
+            throw new Error('attribute "displayKey" can not be empty if input item has "object" type');
+        }
+        const value: string | Any[] = getValueByPath<T, string>(item, displayKey, '') ?? '';
+        if (Array.isArray(value)) {
+            return '';
+        } else {
+            return value;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mansurishmuhametov) [reply packages/common/pipes/src/display-item/display-item.pipe.ts:29]: the code looks cleaner without if/else

**Review by splincode — APPROVED** [review]

**Pull request merged**